packages/build/src/plugins_core/deploy/buildbot_client.ts

```diff
@@ -0,0 +1,167 @@
+import net, { NetConnectOpts } from 'net'
```

Comment:
For some reason GH's diffing didn't kick in for this file, but most of this has been a move from JS -> TS, there's no functionality change besides some bug fixes in our error handling and further error segmentation.